nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py

```diff
@@ -21,20 +21,111 @@
 https://github.com/google-research/google-research/blob/master/schema_guided_dst/baseline/pred_utils.py
 """
 
-import collections
 import json
 import os
+from collections import OrderedDict, defaultdict
 
 from nemo import logging
-from nemo.collections.nlp.data.datasets.sgd_dataset.input_example import STATUS_ACTIVE, STATUS_DONTCARE, STR_DONTCARE
+from nemo.collections.nlp.data.datasets.sgd_dataset.input_example import (
+    STATUS_ACTIVE,
+    STATUS_DONTCARE,
+    STATUS_OFF,
+    STR_DONTCARE,
+)
 
 REQ_SLOT_THRESHOLD = 0.5
+MIN_SLOT_RELATION = 25
 
 __all__ = ['get_predicted_dialog_baseline', 'write_predictions_to_file']
 
 
-def get_predicted_dialog_ret_sys_act(dialog, all_predictions, schemas, eval_debug, in_domain_services):
-    """Update labels in a dialogue based on model predictions.
+def carry_over_slots(
+    cur_usr_frame,
+    all_slot_values,
+    slots_relation_list,
+    prev_frame_service,
+    slot_values,
+    sys_prev_slots,
+    last_sys_slots,
+):
+    """This function searches the candidate list for cross-service cases to find and update the values for all the slots in the current predicted state
+    It is called when state is predicted for a frame.
+    MIN_SLOT_RELATION specifies the minimum number of relations between two slots in the training dialogues to get considered for carry-over
+    """
```

Comment:
all args and returns should also be described

Reply:
Added.